device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1161,7 +1161,6 @@ private void reconnectDeviceSession(TransportException transportException, Strin
         while (this.deviceConnectionStates.get(deviceId) == IotHubConnectionStatus.DISCONNECTED_RETRYING
                 && !hasReconnectOperationTimedOut)
         {
-            log.trace("Attempting to reconnect device session: attempt {}", reconnectionAttempts);
```

Comment:
The next lines of code determine if a retry attempt should even be made. It makes more sense to move this line after that decision is made so that it doesn't appear the SDK is retrying when it actually isn't